### PR TITLE
feat: step block analytics

### PR DESCRIPTION
**Pull request by levi-zustiak**
# Description

### Issue
Need the analytics for a step block

[Link to Basecamp Todo](https://3.basecamp.com/3105655/projects)

### Solution
Added overlay for step block analytics and analytics for the action buttons on a step

# External Changes

# Additional information

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx [hunk 1]
```diff
@@ -20,16 +22,40 @@ export function StepBlockNode({
...
+    (step) => step.stepId === id
+  )
+  const actionBlocks = useMemo(
+    () => (step != null ? [step, ...filterActionBlocks(step)] : []),
+    [step]
+  )
+
+  const blockAnalyticsMap = {}
+
+  // const { blockAnalyticsMap } = useMemo(() => {
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx:37]: Remove comments

**Review by csiyang — CHANGES_REQUESTED** [review]
Are the action bars meant to be here too, or will they be on a different PR?

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/AnalyticsDataPoint/AnalyticsDataPoint.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import Stack from '@mui/material/Stack'
+import SvgIcon from '@mui/material/SvgIcon'
+import Tooltip from '@mui/material/Tooltip'
+import Typography from '@mui/material/Typography'
+import { ReactElement } from 'react'
+
+interface StatsOverlayDataPointProps {
+  value: string | number
+  Icon?: typeof SvgIcon
+  tooltipLabel: string
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/AnalyticsDataPoint/AnalyticsDataPoint.tsx:10]:
```suggestion
  tooltipTitle: string
```

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/AnalyticsDataPoint/AnalyticsDataPoint.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+export function AnalyticsDataPoint({
+  value,
+  Icon,
+  tooltipLabel = ''
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/AnalyticsDataPoint/AnalyticsDataPoint.tsx:16]:
```suggestion
  tooltipTitle
```
not nullable.

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx [hunk 4]
```diff
@@ -20,16 +22,40 @@ export function StepBlockNode({
...
 
   const isSelected =
     activeContent === ActiveContent.Canvas && selectedStep?.id === step?.id
 
   return step != null ? (
-    <>
+    <Stack sx={{ position: 'relative' }}>
+      <Fade in={showAnalytics === true}>
+        <div>
+          <StepBlockNodeAnalytics {...stepStats} />
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx:56]: can this component handle its analytics fetching?

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx [hunk 5]
```diff
@@ -61,20 +87,20 @@ export function StepBlockNode({
...
         </BaseNode>
-        <Stack direction="column">
+        <Stack direction="column" sx={{}}>
           {actionBlocks.map((block) => (
-            <ActionButton key={block.id} block={block} selected={isSelected} />
+            <ActionButton
+              key={block.id}
+              block={block}
+              selected={isSelected}
+              analytics={getBlockAnalytics(block)}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNode.tsx:98]: action button should handle the conversion to analytics.

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/ActionButton/ActionButton.tsx [hunk 6]
```diff
@@ -69,6 +70,23 @@ export function ActionButton({
...
+
+    let percentage = blockEventTotal / (stepEventAnalytics?.total ?? 0)
+
+    if (Number.isNaN(percentage) || !Number.isFinite(percentage)) {
+      percentage = 0
+    }
+
+    return { percentage, total: blockEventTotal }
+  }
+  const { percentage, total } = getBlockAnalytics(block)
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/ActionButton/ActionButton.tsx:88]:
```suggestion
  const total = analytics?.blockMap.get(block.id) ?? 0]
  let percentage = blockEventTotal / (analytics?.stepMap.get(stepId)?.total ?? 0)

  if (Number.isNaN(percentage) || !Number.isFinite(percentage))
    percentage = 0
```

**Review by tataihono — CHANGES_REQUESTED** [review]

**Review by tataihono — APPROVED** [review]

**Review by csiyang — APPROVED** [review]

**Pull request merged**